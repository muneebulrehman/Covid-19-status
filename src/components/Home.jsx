import React from 'react';
import { useSelector } from 'react-redux';
import Card from './Card';
import Header from './Header';
import style from './styling/Home.module.css';

const Home = () => {
  const AllData = useSelector(({ countryReducer }) => countryReducer);
  // console.log(AllData);
  const region = ['Africa', 'Americas', 'Antarctic', 'Asia', 'Europe', 'Oceania'];

  return (
    <div>
      <Header />
      <div className={style.allCards}>
        {region.map((el, i) => (
          <Card
            key={el}
            name={el.toLowerCase()}
            homeCard={true}
            even={i % 2 === 0 ? true : false}
            count={
              AllData.length > 0 &&
              AllData.reduce((acc, cur) => {
                if (cur.region === el) acc += 1;
                return acc;
              }, 0)
            }
          />
        ))}
      </div>
    </div>
  );
};

export default Home;
