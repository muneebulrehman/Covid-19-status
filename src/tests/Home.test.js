import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from '../components/Home';
import store from '../redux/configureStore';

describe('Testing the Home page', () => {
  it('Home page rendered correctly', () => {
    const home = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Home />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(home).toMatchSnapshot();
  });
});
