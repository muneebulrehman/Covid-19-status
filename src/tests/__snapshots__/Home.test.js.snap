// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home page Home page rendered correctly 1`] = `
<div>
  <div
    className="fullSpinner"
  >
    <div
      className="spinner"
    >
      <div
        className="loader l1"
      />
      <div
        className="loader l2"
      />
    </div>
  </div>
</div>
`;
